src/main/java/at/hannibal2/skyhanni/data/MaxwellAPI.kt

```diff
@@ -0,0 +1,120 @@
+
+object MaxwellAPI {
+    val config = ProfileStorageData.profileSpecific
+    var currentPower: String? = null
+    var magicalPower = -1
+    var powers = mutableListOf<String>()
+
+    private val group = RepoPattern.group("data.maxwell")
+    private val chatPowerpattern by group.pattern(
+        "chat.power",
+        "§eYou selected the §a(?<power>.*) §e(power )?for your §aAccessory Bag§e!"
+    )
+    private val inventoryPowerPattern by group.pattern(
+        "inventory.power",
+        "§7Selected Power: §a(?<power>.*)"
+    )
+    private val inventoryMPPattern by group.pattern(
+        "inventory.mp",
+        "§7Magical Power: §6(?<mp>[\\d,]+)"
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        val message = event.message.trimWhiteSpaceAndResets().removeResets()
+
+        chatPowerpattern.matchMatcher(message) {
+            val power = group("power")
+            currentPower = byNameOrNull(power) ?: return
+            savePower(currentPower)
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+
+        if (event.inventoryName.contains("Accessory Bag Thaumaturgyr")) {
+            val stacks = event.inventoryItems
+            val selectedPower =
+                stacks.values.find { it.getLore().isNotEmpty() && it.getLore().last() == "§aPower is selected!" }
+                    ?: return
+
+            currentPower = byNameOrNull(selectedPower.getLore().first()) ?: return
+            savePower(currentPower)
+
+        } else if (event.inventoryName.contains("Your Bags")) {
+            val stacks = event.inventoryItems
+
+            for (stack in stacks.values) {
+                val lore = stack.getLore()
+                for (line in lore) {
+                    inventoryPowerPattern.matchMatcher(line) {
+                        val power = group("power")
+                        currentPower = byNameOrNull(power) ?: return
+                        savePower(currentPower)
+                    }
+                    inventoryMPPattern.matchMatcher(line) {
+                        // Mp is boosted in catacombs
+                        if (IslandType.CATACOMBS.isInIsland()) return
+
+                        val mp = group("mp")
+                        magicalPower = mp.replace(",", "").toIntOrNull() ?: return
+                        saveMP(magicalPower)
+                    }
+                }
+            }
+        }
+    }
+
+    fun byNameOrNull(name: String) = powers.find { it == name }
+
+    // Handle storage
+    @SubscribeEvent
+    fun onRepoLoad(event: RepositoryReloadEvent) {
+        val data = event.getConstant<MaxwellPowersJson>("MaxwellPowers")
+        powers = data.powers
+    }
+
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        val config = ProfileStorageData.profileSpecific ?: return
```

Comment:
change those lines to this since config is a get now
```suggestion
        if(config == null ) return
```